### ISSUE TITLE
Collision layer bit over 32 or negative can be set in `3.x`

### ISSUE THREAD

**Issue by RikKargones** [godotengine/godot#49934]
### Godot version

3.3.2 stable

### System information

Windows 7

### Issue description

Issue in topic. I think engine should trow an error, if it can't support what layer bit anyway. It might be important for some users, for example, what can choose exact layer bit by formula or something.

And just little request for addition: please, make shown in editor all supported 32 collision layer bits. It's was an unpleasant surprise for me and give addition unnecessary work to rewriting code. At least just make a short-term solution, if you [discuss about UI usability in editor](https://github.com/godotengine/godot-proposals/issues/2770), it will be better than nothing.

### Steps to reproduce

1. Add any PhysicsBody2D or Area2D to scene, add script to them.
2. Set by _set_collision_layer_bit_ any unsupported bit (-1, 37, 96 e.t.c) to **true** in code.
3. Ready to check and show! Code for checking may looks like what:
```
for i in 70:
    if get_collision_layer_bit(i-10): print(i-10)
```

**Comment by greatmomo** [comment]
Interestingly enough, the values seem to be added twice when they exceed the limits.
For example, ``set_collision_layer_bit(36, true)`` sets **36**, but also **4** (36-32).

**Comment by Calinou** [comment]
This was fixed in the `master` branch by https://github.com/godotengine/godot/pull/34823, but was never cherry-picked or remade for the `3.x` branch.

**Comment by akien-mga** [comment]
Fixed by #34823 (4.0) and #49955 (3.4).
